From 847c8d255d2fe8f8ff66a3221a3469c2c3f3aab7 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 29 May 2014 23:28:01 -0700
Subject: [PATCH] Add @asyncio.coroutine to a ton of plugins

Note: also updates some plugins to new format (inp -> text, multiple @hook.commands -> one hook)

diff --git a/plugins/admin.py b/plugins/admin.py
index dd288e5..8562c3a 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -1,8 +1,10 @@
+import asyncio
 import re
 
 from cloudbot import hook
 
 
+@asyncio.coroutine
 @hook.command(["groups", "listgroups", "permgroups"], permissions=["permissions_users"], autohelp=False)
 def get_permission_groups(conn):
     """groups -- lists all valid groups
@@ -11,6 +13,7 @@ def get_permission_groups(conn):
     return "Valid groups: {}".format(conn.permissions.get_groups())
 
 
+@asyncio.coroutine
 @hook.command("gperms", permissions=["permissions_users"])
 def get_group_permissions(text, conn, notice):
     """gperms <group> -- lists permissions of a group
@@ -29,6 +32,7 @@ def get_group_permissions(text, conn, notice):
         notice("Unknown group '{}'".format(group))
 
 
+@asyncio.coroutine
 @hook.command("gusers", permissions=["permissions_users"])
 def get_group_users(text, conn, notice):
     """gusers <group> -- lists users in a group
@@ -47,6 +51,7 @@ def get_group_users(text, conn, notice):
         notice("Unknown group '{}'".format(group))
 
 
+@asyncio.coroutine
 @hook.command("uperms", autohelp=False)
 def get_user_permissions(text, conn, mask, has_permission, notice):
     """uperms [user] -- lists all permissions given to a user, or the current user if none is given
@@ -71,6 +76,7 @@ def get_user_permissions(text, conn, mask, has_permission, notice):
         return "User {} has no elevated permissions".format(user)
 
 
+@asyncio.coroutine
 @hook.command("ugroups", autohelp=False)
 def get_user_groups(text, conn, mask, has_permission, notice):
     """uperms [user] -- lists all permissions given to a user, or the current user if none is given
@@ -95,6 +101,7 @@ def get_user_groups(text, conn, mask, has_permission, notice):
         return "User {} is in no permission groups".format(user)
 
 
+@asyncio.coroutine
 @hook.command("deluser", permissions=["permissions_users"])
 def remove_permission_user(text, bot, conn, notice, reply):
     """deluser <user> [group] -- Removes a user from a permission group, or all permission groups if none is specified
@@ -150,6 +157,7 @@ def remove_permission_user(text, bot, conn, notice, reply):
         permission_manager.reload()
 
 
+@asyncio.coroutine
 @hook.command("adduser", permissions=["permissions_users"])
 def add_permissions_user(text, conn, bot, notice, reply):
     """adduser <user> <group> -- Adds a user to a permission group
@@ -215,6 +223,7 @@ def restart(text, bot):
         bot.restart()
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def join(text, conn, notice):
     """join <channel> -- Joins a given channel
@@ -228,6 +237,7 @@ def join(text, conn, notice):
         conn.join(target)
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"], autohelp=False)
 def part(text, conn, chan, notice):
     """part [channel] -- Leaves a given channel, or the current one if no channel is specified
@@ -246,6 +256,7 @@ def part(text, conn, chan, notice):
         conn.part(target)
 
 
+@asyncio.coroutine
 @hook.command(autohelp=False, permissions=["botcontrol"])
 def cycle(text, conn, chan, notice):
     """cycle <channel> -- Cycles a given channel, or the current one if no channel is specified
@@ -265,6 +276,7 @@ def cycle(text, conn, chan, notice):
         conn.join(target)
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def nick(text, conn, notice):
     """nick <nick> -- Changes the bot's nickname
@@ -278,6 +290,7 @@ def nick(text, conn, notice):
     conn.set_nick(text)
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def raw(text, conn, notice):
     """raw <command> -- Sends a irc_raw IRC command
@@ -288,6 +301,7 @@ def raw(text, conn, notice):
     conn.send(text)
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def say(text, conn, chan):
     """say [channel] <message> -- Makes the bot say <message> in [channel], or the current channel if none is specified
@@ -306,6 +320,7 @@ def say(text, conn, chan):
     conn.msg(channel, text)
 
 
+@asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def message(text, conn):
     """message <name> <message> -- Makes the bot say <message> to <name>, <name> may be a #channel or a nickname
@@ -318,8 +333,8 @@ def message(text, conn):
     conn.msg(channel, text)
 
 
-@hook.command("act", permissions=["botcontrol"])
-@hook.command(permissions=["botcontrol"])
+@asyncio.coroutine
+@hook.command(["me", "act"], permissions=["botcontrol"])
 def me(text, conn, chan):
     """me [channel] <action> -- Makes the bot act out <action> in a [channel], or the current channel if none is given
     :type text: str
diff --git a/plugins/attacks.py b/plugins/attacks.py
index f4967d6..00edafe 100644
--- a/plugins/attacks.py
+++ b/plugins/attacks.py
@@ -1,4 +1,5 @@
 import random
+import asyncio
 import re
 
 from cloudbot import hook
@@ -27,7 +28,8 @@ def is_self(conn, target):
         return False
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def lart(text, conn, nick, notice, action):
     """lart <user> -- LARTs <user>.
     :type text: str
@@ -50,7 +52,8 @@ def lart(text, conn, nick, notice, action):
     action(phrase.format(user=target))
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def insult(text, conn, nick, notice, message):
     """insult <user> -- Makes the bot insult <user>.
     :type text: str
@@ -70,7 +73,8 @@ def insult(text, conn, nick, notice, message):
     message("{}, {}".format(target, random.choice(insults)))
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def flirt(text, conn, nick, notice, message):
     """flirt <user> -- Makes the bot flirt with <user>.
     :type text: str
diff --git a/plugins/brainfuck.py b/plugins/brainfuck.py
index 428f8b0..c73cecf 100644
--- a/plugins/brainfuck.py
+++ b/plugins/brainfuck.py
@@ -2,6 +2,7 @@
 http://brainfuck.sourceforge.net/brain.py"""
 
 import re
+import asyncio
 import random
 
 from cloudbot import hook
@@ -10,6 +11,7 @@ BUFFER_SIZE = 5000
 MAX_STEPS = 1000000
 
 
+@asyncio.coroutine
 @hook.command(["brainfuck", "bf"])
 def bf(text):
     """bf <prog> -- Executes <prog> as Brainfuck code.
diff --git a/plugins/choose.py b/plugins/choose.py
index 06f0f70..c3ce1ba 100644
--- a/plugins/choose.py
+++ b/plugins/choose.py
@@ -1,10 +1,12 @@
 import re
+import asyncio
 import random
 
 from cloudbot import hook
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def choose(text, notice):
     """choose <choice1>, [choice2], [choice3], etc. -- Randomly picks one of the given choices.
     :type text: str
diff --git a/plugins/coin.py b/plugins/coin.py
index cfbf144..398402a 100644
--- a/plugins/coin.py
+++ b/plugins/coin.py
@@ -1,8 +1,10 @@
+import asyncio
 import random
 
 from cloudbot import hook
 
 
+@asyncio.coroutine
 @hook.command(autohelp=False)
 def coin(text, notice, action):
     """coin [amount] -- Flips [amount] of coins.
diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 8453d57..6ef1828 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -1,19 +1,23 @@
+import asyncio
 import time
 
 from cloudbot import hook
 
 
 # CTCP responses
+@asyncio.coroutine
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(notice):
     notice('\x01VERSION Eliralin #refresh - http://git.io/elirefresh')
 
 
+@asyncio.coroutine
 @hook.regex(r'^\x01PING\x01$')
 def ctcp_ping(notice):
     notice('\x01PING: PONG')
 
 
+@asyncio.coroutine
 @hook.regex(r'^\x01TIME\x01$')
 def ctcp_time(notice):
     notice('\x01TIME: The time is: {}'.format(time.strftime("%r", time.localtime())))
diff --git a/plugins/core_misc.py b/plugins/core_misc.py
index a296c88..e5d1892 100644
--- a/plugins/core_misc.py
+++ b/plugins/core_misc.py
@@ -7,6 +7,7 @@ socket.setdefaulttimeout(10)
 
 
 # Auto-join on Invite (Configurable, defaults to True)
+@asyncio.coroutine
 @hook.event('INVITE')
 def invite(paramlist, conn):
     """
@@ -19,7 +20,8 @@ def invite(paramlist, conn):
 
 
 # Identify to NickServ (or other service)
-@hook.event('004', threaded=False)
+@asyncio.coroutine
+@hook.event('004')
 def onjoin(conn, bot):
     """
     :type conn: cloudbot.core.connection.BotConnection
@@ -58,7 +60,8 @@ def onjoin(conn, bot):
     bot.logger.info("ONJOIN hook completed. Bot ready.")
 
 
-@hook.event('004', threaded=False)
+@asyncio.coroutine
+@hook.event('004')
 def keep_alive(conn):
     """
     :type conn: core.irc.BotConnection
diff --git a/plugins/core_tracker.py b/plugins/core_tracker.py
index 6290333..4b8fa7a 100644
--- a/plugins/core_tracker.py
+++ b/plugins/core_tracker.py
@@ -1,5 +1,6 @@
 # plugin to keep track of bot state
 
+import asyncio
 import re
 
 from cloudbot import hook
@@ -7,6 +8,7 @@ from cloudbot import hook
 nick_re = re.compile(":(.+?)!")
 
 
+@asyncio.coroutine
 @hook.event("KICK")
 def on_kick(irc_paramlist, conn, chan):
     """
@@ -22,6 +24,7 @@ def on_kick(irc_paramlist, conn, chan):
             conn.join(irc_paramlist[0])
 
 
+@asyncio.coroutine
 @hook.event("NICK")
 def on_nick(irc_paramlist, bot, conn, irc_raw):
     """
diff --git a/plugins/correction.py b/plugins/correction.py
index 8acae05..0015c5c 100644
--- a/plugins/correction.py
+++ b/plugins/correction.py
@@ -1,3 +1,4 @@
+import asyncio
 import re
 
 from cloudbot import hook
@@ -5,6 +6,7 @@ from cloudbot import hook
 correction_re = re.compile(r"^[sS]/([^/]*)/([^/]*)(/.*)?\s*$")
 
 
+@asyncio.coroutine
 @hook.regex(correction_re)
 def correction(match, conn, chan, message):
     """
diff --git a/plugins/cypher.py b/plugins/cypher.py
index 3c45b5e..4052c27 100644
--- a/plugins/cypher.py
+++ b/plugins/cypher.py
@@ -1,4 +1,5 @@
 import base64
+import asyncio
 import binascii
 
 from cloudbot import hook
@@ -35,7 +36,8 @@ def decode(password, encoded, notice):
     return "".join(dec)
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def cypher(text, notice):
     """cypher <pass> <string> -- Cyphers <string> with <password>.
     :type text: str
@@ -49,7 +51,8 @@ def cypher(text, notice):
     return encode(password, plaintext)
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def decypher(text, notice):
     """decypher <pass> <string> -- Decyphers <string> with <password>.
     :type text: str
diff --git a/plugins/dbtest.py b/plugins/dbtest.py
index 46024dd..53bd0da 100644
--- a/plugins/dbtest.py
+++ b/plugins/dbtest.py
@@ -9,7 +9,7 @@ users = Table(
 )
 
 
-@hook.command
+@hook.command()
 def dbadduser(text, db):
     """
     :type text: str
diff --git a/plugins/dice.py b/plugins/dice.py
index e12c218..a380992 100644
--- a/plugins/dice.py
+++ b/plugins/dice.py
@@ -2,6 +2,7 @@
 
 
 import re
+import asyncio
 import random
 
 from cloudbot import hook
@@ -34,6 +35,7 @@ def n_rolls(count, n):
 
 
 #@hook.regex(valid_diceroll, re.I)
+@asyncio.coroutine
 @hook.command(["roll", "dice"])
 def dice(text, notice):
     """dice <dice roll> -- Simulates dice rolls. Example: 'dice 2d20-d5+4 roll 2': D20s, subtract 1D5, add 4
diff --git a/plugins/drama.py b/plugins/drama.py
index 76ee990..ac7940e 100644
--- a/plugins/drama.py
+++ b/plugins/drama.py
@@ -7,7 +7,7 @@ api_url = "http://encyclopediadramatica.se/api.php?action=opensearch"
 ed_url = "http://encyclopediadramatica.se/"
 
 
-@hook.command
+@hook.command()
 def drama(text):
     """drama <phrase> -- Gets the first paragraph of
     the Encyclopedia Dramatica article on <phrase>."""
diff --git a/plugins/eightball.py b/plugins/eightball.py
index d4b1062..2dc8073 100644
--- a/plugins/eightball.py
+++ b/plugins/eightball.py
@@ -1,4 +1,5 @@
 import os
+import asyncio
 import random
 
 from cloudbot import hook, formatting
@@ -19,6 +20,7 @@ def load_responses(bot):
                      f.readlines() if not line.startswith("//")]
 
 
+@asyncio.coroutine
 @hook.command(["8ball", "8", "eightball"])
 def eightball(action):
     """8ball <question> -- The all knowing magic eight ball, in electronic form. Ask and it shall be answered!"""
diff --git a/plugins/encrypt.py b/plugins/encrypt.py
index 21a9d45..c6f1d5e 100644
--- a/plugins/encrypt.py
+++ b/plugins/encrypt.py
@@ -1,6 +1,7 @@
 import os
 import base64
 import hashlib
+import asyncio
 import traceback
 
 from pbkdf2 import PBKDF2
@@ -60,7 +61,8 @@ def get_salt(bot):
     return bot.config.get("random_salt")
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def encrypt(text, bot, db, notice):
     """encrypt <pass> <string> -- Encrypts <string> with <pass>. (<string> can only be decrypted using this bot)
     :type text: str
@@ -98,7 +100,8 @@ def encrypt(text, bot, db, notice):
     return encoded.decode()
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def decrypt(text, bot, db, notice):
     """decrypt <pass> <string> -- Decrypts <string> with <pass>. (can only decrypt strings encrypted on this bot)
     :type bot: core.bot.CloudBot
diff --git a/plugins/factoids.py b/plugins/factoids.py
index a7cfe07..7fffe41 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -1,5 +1,6 @@
 # Written by Scaevolus 2010
 import string
+import asyncio
 import re
 
 from sqlalchemy import Table, Column, String
@@ -27,6 +28,7 @@ table = Table(
     Column("nick", String)
 )
 
+
 @hook.onload()
 def load_cache(db):
     """
@@ -68,6 +70,7 @@ def del_factoid(db, word):
     load_cache(db)
 
 
+@asyncio.coroutine
 @hook.command(["r", "remember"], permissions=["addfactoid"])
 def remember(text, nick, db, notice):
     """remember <word> [+]<data> -- Remembers <data> with <word>. Add + to <data> to append."""
@@ -118,7 +121,8 @@ def forget(text, db, notice):
         return
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def info(text, notice):
     """info <factoid> -- Shows the source of a factoid."""
 
@@ -130,6 +134,7 @@ def info(text, notice):
         notice("Unknown Factoid.")
 
 
+@asyncio.coroutine
 @hook.regex(r'^\^ ?(.+)')
 def factoid(inp, input, db, message, action):
     """?<word> -- Shows what data is associated with <word>."""
@@ -171,6 +176,7 @@ def factoid(inp, input, db, message, action):
             message(result)
 
 
+@asyncio.coroutine
 @hook.command(autohelp=False, permissions=["listfactoids"])
 def listfactoids(reply):
     reply_text = []
diff --git a/plugins/fmylife.py b/plugins/fmylife.py
index 34e4e86..593d8d1 100644
--- a/plugins/fmylife.py
+++ b/plugins/fmylife.py
@@ -1,4 +1,5 @@
 from bs4 import BeautifulSoup
+import asyncio
 import requests
 
 from cloudbot import hook
@@ -24,7 +25,7 @@ def initial_refresh():
     refresh_cache()
 
 
-@hook.async
+@asyncio.coroutine
 @hook.command(autohelp=False)
 def fml(reply, loop):
     """fml -- Gets a random quote from fmyfife.com."""
diff --git a/plugins/fortune.py b/plugins/fortune.py
index 1415d1d..df84421 100644
--- a/plugins/fortune.py
+++ b/plugins/fortune.py
@@ -14,8 +14,8 @@ def load_fortunes(bot):
                     if not line.startswith("//")]
 
 
-@hook.command(autohelp=False, threaded=False)
 @asyncio.coroutine
+@hook.command(autohelp=False)
 def fortune():
     """fortune -- Fortune cookies on demand."""
     return random.choice(fortunes)
diff --git a/plugins/geoip.py b/plugins/geoip.py
index 7156d52..921397a 100644
--- a/plugins/geoip.py
+++ b/plugins/geoip.py
@@ -32,7 +32,7 @@ def load_regions(bot):
         geo = pygeoip.GeoIP(os.path.join(bot.data_dir, "GeoLiteCity.dat"))
 
 
-@hook.command
+@hook.command()
 def geoip(text):
     """geoip <host/ip> -- Gets the location of <host/ip>"""
 
diff --git a/plugins/google_translate.py b/plugins/google_translate.py
index d48e092..6c10ab2 100644
--- a/plugins/google_translate.py
+++ b/plugins/google_translate.py
@@ -73,7 +73,7 @@ def match_language(fragment):
     return None
 
 
-@hook.command
+@hook.command()
 def translate(text, bot):
     """translate [source language [target language]] <sentence> -- translates
     <sentence> from source language (default autodetect) to target
diff --git a/plugins/help.py b/plugins/help.py
index de81f6e..cc8a6a8 100644
--- a/plugins/help.py
+++ b/plugins/help.py
@@ -1,9 +1,11 @@
 from operator import attrgetter
+import asyncio
 import re
 
 from cloudbot import hook
 
 
+@asyncio.coroutine
 @hook.command("help", autohelp=False)
 def help_command(text, conn, bot, notice, has_permission):
     """help  -- Gives a list of commands/help for a command.
diff --git a/plugins/history.py b/plugins/history.py
index 41466bd..add40d0 100644
--- a/plugins/history.py
+++ b/plugins/history.py
@@ -1,5 +1,6 @@
 from collections import deque
 import time
+import asyncio
 import re
 
 from cloudbot import hook, timesince
@@ -49,6 +50,7 @@ def chat_tracker(event, db, conn):
     track_history(event, message_time, conn)
 
 
+@asyncio.coroutine
 @hook.command(autohelp=False)
 def resethistory(event, conn):
     """resethistory - Resets chat history for the current channel"""
diff --git a/plugins/horoscope.py b/plugins/horoscope.py
index b6c1253..394cb49 100644
--- a/plugins/horoscope.py
+++ b/plugins/horoscope.py
@@ -3,7 +3,7 @@
 from cloudbot import hook, http, formatting
 
 
-@hook.onload
+@hook.onload()
 def init(db):
     db.execute("create table if not exists horoscope(nick primary key, sign)")
     db.commit()
diff --git a/plugins/ignore.py b/plugins/ignore.py
index 00d34b3..c324d33 100644
--- a/plugins/ignore.py
+++ b/plugins/ignore.py
@@ -1,3 +1,4 @@
+import asyncio
 from fnmatch import fnmatch
 
 from cloudbot import hook
@@ -21,7 +22,8 @@ def ensure_ignored(bot):
         bot.config.save_config()
 
 
-@hook.sieve
+@asyncio.coroutine
+@hook.sieve()
 def ignore_sieve(bot, event, _hook):
     """ blocks events from ignored channels/hosts
     :type bot: cloudbot.core.bot.CloudBot
@@ -50,6 +52,7 @@ def ignore_sieve(bot, event, _hook):
     return event
 
 
+@asyncio.coroutine
 @hook.command(autohelp=False)
 def ignored(notice, conn):
     """ignored -- Lists ignored channels/users."""
diff --git a/plugins/imdb.py b/plugins/imdb.py
index e63add5..20a05a7 100644
--- a/plugins/imdb.py
+++ b/plugins/imdb.py
@@ -8,7 +8,7 @@ id_re = re.compile("tt\d+")
 imdb_re = (r'(.*:)//(imdb.com|www.imdb.com)(:[0-9]+)?(.*)', re.I)
 
 
-@hook.command
+@hook.command()
 def imdb(text):
     """imdb <movie> -- Gets information about <movie> from IMDb."""
 
diff --git a/plugins/kill.py b/plugins/kill.py
index d418efd..9c491f9 100644
--- a/plugins/kill.py
+++ b/plugins/kill.py
@@ -1,3 +1,4 @@
+import asyncio
 import json
 
 from cloudbot import hook, textgen
@@ -9,7 +10,8 @@ def get_generator(_json, variables):
                                  data["parts"], variables=variables)
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def kill(text, action=None, nick=None, conn=None, notice=None):
     """kill <user> -- Makes the bot kill <user>."""
     target = text.strip()
diff --git a/plugins/lastfm.py b/plugins/lastfm.py
index 8d0d14b..27e16ea 100644
--- a/plugins/lastfm.py
+++ b/plugins/lastfm.py
@@ -5,21 +5,19 @@ from cloudbot import hook, http, timesince
 api_url = "http://ws.audioscrobbler.com/2.0/?format=json"
 
 
-@hook.command('l', autohelp=False)
-@hook.command(autohelp=False)
-def lastfm(inp, nick='', db=None, bot=None, notice=None):
-    """lastfm [user] [dontsave] -- Displays the now playing (or last played)
-     track of LastFM user [user]."""
+@hook.command(["lastfm", "l"], autohelp=False)
+def lastfm(text, nick, db, bot, notice):
+    """lastfm [user] [dontsave] -- Displays the now playing (or last played) track of LastFM user [user]."""
     api_key = bot.config.get("api_keys", {}).get("lastfm")
     if not api_key:
         return "error: no api key set"
 
     # check if the user asked us not to save his details
-    dontsave = inp.endswith(" dontsave")
+    dontsave = text.endswith(" dontsave")
     if dontsave:
-        user = inp[:-9].strip().lower()
+        user = text[:-9].strip().lower()
     else:
-        user = inp
+        user = text
 
     db.execute("create table if not exists lastfm(nick primary key, acc)")
 
@@ -74,9 +72,9 @@ def lastfm(inp, nick='', db=None, bot=None, notice=None):
     # append ending based on what type it was
     out += ending
 
-    if inp and not dontsave:
-        db.execute("insert or replace into lastfm(nick, acc) values "
-                   "(:nick, :account)", {'nick': nick.lower(), 'account': user})
+    if text and not dontsave:
+        db.execute("insert or replace into lastfm(nick, acc) values (:nick, :account)",
+                   {'nick': nick.lower(), 'account': user})
         db.commit()
 
     return out
diff --git a/plugins/log.py b/plugins/log.py
index b9158f4..df44215 100644
--- a/plugins/log.py
+++ b/plugins/log.py
@@ -3,6 +3,7 @@ log.py: written by Scaevolus 2009
 
 edited 2014
 """
+import asyncio
 
 import os
 import codecs
@@ -137,7 +138,8 @@ def log(event):
 
 
 # Log console separately to prevent lag
-@hook.event("*", threaded=False)
+@asyncio.coroutine
+@hook.event("*")
 def console_log(bot, event):
     """
     :type bot: cloudbot.core.bot.CloudBot
diff --git a/plugins/lyrics.py b/plugins/lyrics.py
index a9e6152..50682a5 100644
--- a/plugins/lyrics.py
+++ b/plugins/lyrics.py
@@ -4,14 +4,14 @@ url = "http://search.azlyrics.com/search.php?q="
 
 
 @hook.command
-def lyrics(inp):
+def lyrics(text):
     """lyrics <search> - Search AZLyrics.com for song lyrics"""
-    if "pastelyrics" in inp:
+    if "pastelyrics" in text:
         dopaste = True
-        inp = inp.replace("pastelyrics", "").strip()
+        text = text.replace("pastelyrics", "").strip()
     else:
         dopaste = False
-    soup = http.get_soup(url + inp.replace(" ", "+"))
+    soup = http.get_soup(url + text.replace(" ", "+"))
     if "Try to compose less restrictive search query" in soup.find('div', {'id': 'inn'}).text:
         return "No results. Check spelling."
     div = None
@@ -40,4 +40,4 @@ def lyrics(inp):
         return "\x02{}\x02 by \x02{}\x02 {}{} - {}".format(title, artist, web.try_shorten(link), pasteurl,
                                                            lyricsum[:-3])
     else:
-        return "No song results. " + url + inp.replace(" ", "+")
+        return "No song results. " + url + text.replace(" ", "+")
diff --git a/plugins/metacritic.py b/plugins/metacritic.py
index 0b5adf0..71dbf35 100644
--- a/plugins/metacritic.py
+++ b/plugins/metacritic.py
@@ -6,8 +6,7 @@ from urllib.error import HTTPError
 from cloudbot import hook, http
 
 
-@hook.command('mc')
-@hook.command
+@hook.command(["metacritic", "mc"])
 def metacritic(inp):
     """mc [all|movie|tv|album|x360|ps3|pc|gba|ds|3ds|wii|vita|wiiu|xone|ps4] <title>
     Gets rating for <title> from metacritic on the specified medium."""
diff --git a/plugins/minecraft_bukget.py b/plugins/minecraft_bukget.py
index dd4a1c5..1dfe9d8 100644
--- a/plugins/minecraft_bukget.py
+++ b/plugins/minecraft_bukget.py
@@ -119,8 +119,7 @@ def format_output(data):
 
 ## HOOK FUNCTIONS
 
-@hook.command('plugin')
-@hook.command
+@hook.command(["bukget", "plugin"])
 def bukget(text, reply, message):
     """bukget <slug/name> - Look up a plugin on dev.bukkit.org"""
     # get the plugin slug using search
diff --git a/plugins/minecraft_items.py b/plugins/minecraft_items.py
index 03ef37b..e20b075 100644
--- a/plugins/minecraft_items.py
+++ b/plugins/minecraft_items.py
@@ -1,6 +1,7 @@
 """ plugin by _303 (?)
 """
 
+import asyncio
 import re
 
 from cloudbot import hook
@@ -48,6 +49,7 @@ with open("./data/itemids.txt") as f:
         ids.append((itemid, name))
 
 
+@asyncio.coroutine
 @hook.command(["mcitem", "mcid"])
 def mcitem(text, reply):
     """mcitem <item/id> -- gets the id from an item or vice versa"""
@@ -78,6 +80,7 @@ def mcitem(text, reply):
     return out
 
 
+@asyncio.coroutine
 @hook.command(["mccraft", "mcrecipe"])
 def mcrecipe(text, reply):
     """mcrecipe <item> -- gets the crafting recipe for an item"""
diff --git a/plugins/minecraft_wiki.py b/plugins/minecraft_wiki.py
index a8a3b25..1d724f9 100644
--- a/plugins/minecraft_wiki.py
+++ b/plugins/minecraft_wiki.py
@@ -6,7 +6,7 @@ api_url = "http://minecraft.gamepedia.com/api.php?action=opensearch"
 mc_url = "http://minecraft.gamepedia.com/"
 
 
-@hook.command
+@hook.command()
 def mcwiki(text):
     """mcwiki <phrase> -- Gets the first paragraph of
     the Minecraft Wiki article on <phrase>."""
diff --git a/plugins/notes.py b/plugins/notes.py
index 7024186..7ff65e2 100644
--- a/plugins/notes.py
+++ b/plugins/notes.py
@@ -90,12 +90,12 @@ def db_search(db, nick, query):
 
 
 @hook.command(["note", "notes"])
-def note(inp, nick, db, notice):
+def note(text, nick, db, notice):
     """note(s) <add|del|list|search> args -- Manipulates your list of notes."""
 
     db_init(db)
 
-    parts = inp.split()
+    parts = text.split()
     cmd = parts[0].lower()
 
     args = parts[1:]
diff --git a/plugins/prefixes.py b/plugins/prefixes.py
index 52550b3..152d1aa 100644
--- a/plugins/prefixes.py
+++ b/plugins/prefixes.py
@@ -1,3 +1,4 @@
+import asyncio
 import re
 
 from sqlalchemy import Table, Column, String, UniqueConstraint
@@ -66,7 +67,8 @@ def delprefix(text, chan, db):
     return "Removed command prefix {} from {}".format(text, chan)
 
 
-@hook.event("PRIVMSG", threaded=False)
+@asyncio.coroutine
+@hook.event("PRIVMSG")
 def run_extra_prefix(event, bot, conn, chan, irc_message):
     """
     :type event: cloudbot.core.events.BaseEvent
diff --git a/plugins/profiling.py b/plugins/profiling.py
index be201f0..2ed38e2 100644
--- a/plugins/profiling.py
+++ b/plugins/profiling.py
@@ -1,12 +1,14 @@
 import gc
 
+import asyncio
 import objgraph
 from pympler import muppy, summary, tracker
 
 from cloudbot import hook, threaddump
 
 
-@hook.command(threaded=False, autohelp=False, permissions=["botcontrol"])
+@asyncio.coroutine
+@hook.command(autohelp=False, permissions=["botcontrol"])
 def threaddump_command(reply):
     reply("Running thread dump")
     return "Thread dump located at {}".format(threaddump.get_thread_dump())
diff --git a/plugins/snopes.py b/plugins/snopes.py
index 806cd8c..746cc41 100644
--- a/plugins/snopes.py
+++ b/plugins/snopes.py
@@ -6,10 +6,10 @@ search_url = "http://search.atomz.com/search/?sp_a=00062d45-sp00000000"
 
 
 @hook.command
-def snopes(inp):
+def snopes(text):
     """snopes <topic> -- Searches snopes for an urban legend about <topic>."""
 
-    search_page = http.get_html(search_url, sp_q=inp, sp_c="1")
+    search_page = http.get_html(search_url, sp_q=text, sp_c="1")
     result_urls = search_page.xpath("//a[@target='_self']/@href")
 
     if not result_urls:
diff --git a/plugins/steam_calc.py b/plugins/steam_calc.py
index dcd0f4c..07b2e0d 100644
--- a/plugins/steam_calc.py
+++ b/plugins/steam_calc.py
@@ -31,18 +31,16 @@ def unicode_dictreader(utf8_data, **kwargs):
         yield dict([(key.lower(), str(value, 'utf-8')) for key, value in row.items()])
 
 
-@hook.command('sc')
-@hook.command
-def steamcalc(inp, reply=None):
-    """steamcalc <username> [currency] - Gets value of steam account and
-       total hours played. Uses steamcommunity.com/id/<nickname>. """
+@hook.command(["steamcalc", "sc"])
+def steamcalc(text, reply):
+    """steamcalc <username> [currency] - Gets value of steam account and total hours played. Uses steamcommunity.com/id/<nickname>. """
 
     # check if the user asked us to force reload
-    force_reload = inp.endswith(" forcereload")
+    force_reload = text.endswith(" forcereload")
     if force_reload:
-        name = inp[:-12].strip().lower()
+        name = text[:-12].strip().lower()
     else:
-        name = inp.strip()
+        name = text.strip()
 
     if force_reload:
         try:
diff --git a/plugins/tell.py b/plugins/tell.py
index d94ea66..fcd3bf6 100644
--- a/plugins/tell.py
+++ b/plugins/tell.py
@@ -108,9 +108,9 @@ def showtells(nick, notice, db, conn):
 
 
 @hook.command("tell")
-def tell_cmd(inp, nick, db, notice, conn):
+def tell_cmd(text, nick, db, notice, conn):
     """tell <nick> <message> -- Relay <message> to <nick> when <nick> is around."""
-    query = inp.split(' ', 1)
+    query = text.split(' ', 1)
 
     if len(query) != 2:
         notice(conn.config("command_prefix") + tell_cmd.__doc__)
diff --git a/plugins/validate.py b/plugins/validate.py
index bd4bcc3..5da98c2 100644
--- a/plugins/validate.py
+++ b/plugins/validate.py
@@ -7,8 +7,7 @@ by Vladi
 from cloudbot import hook, http
 
 
-@hook.command('w3c')
-@hook.command
+@hook.command(["validate", "w3c"])
 def validate(text):
     """validate <url> -- Runs url through the w3c markup validator."""
 
diff --git a/plugins/valvesounds.py b/plugins/valvesounds.py
index 7684d14..55a27de 100644
--- a/plugins/valvesounds.py
+++ b/plugins/valvesounds.py
@@ -33,60 +33,48 @@ def get_sound_info(game, search):
 
 
 @hook.command
-def portal2(inp):
-    """portal2 <quote> - Look up Portal 2 quote.
-    Example: .portal2 demand to see life's manager"""
-    return get_sound_info("portal2", inp)
+def portal2(text):
+    """portal2 <quote> - Look up Portal 2 quote. Example: .portal2 demand to see life's manager"""
+    return get_sound_info("portal2", text)
 
 
 @hook.command
-def portal2dlc(inp):
-    """portal2dlc <quote> - Look up Portal 2 DLC quote.
-    Example: .portal2dlc1 these exhibits are interactive"""
-    return get_sound_info("portal2dlc1", inp)
+def portal2dlc(text):
+    """portal2dlc <quote> - Look up Portal 2 DLC quote. Example: .portal2dlc1 these exhibits are interactive"""
+    return get_sound_info("portal2dlc1", text)
 
 
-@hook.command("portal2pti")
-@hook.command
-def portal2dlc2(inp):
-    """portal2dlc2 <quote> - Look up Portal 2 Perpetual Testing Inititive quote.
-    Example: .portal2 Cave here."""
-    return get_sound_info("portal2dlc2", inp)
+@hook.command(["portal2dlc2", "portal2pti"])
+def portal2dlc2(text):
+    """portal2dlc2 <quote> - Look up Portal 2 Perpetual Testing Inititive quote. Example: .portal2 Cave here."""
+    return get_sound_info("portal2dlc2", text)
 
 
 @hook.command
-def portal2music(inp):
-    """portal2music <title> - Look up Portal 2 music.
-    Example: .portal2music turret opera"""
-    return get_sound_info("portal2music", inp)
+def portal2music(text):
+    """portal2music <title> - Look up Portal 2 music. Example: .portal2music turret opera"""
+    return get_sound_info("portal2music", text)
 
 
-@hook.command('portal1')
-@hook.command
-def portal(inp):
-    """portal <quote> - Look up Portal quote.
-    Example: .portal The last thing you want to do is hurt me"""
-    return get_sound_info("portal1", inp)
+@hook.command(["portal", "portal1"])
+def portal(text):
+    """portal <quote> - Look up Portal quote. Example: .portal The last thing you want to do is hurt me"""
+    return get_sound_info("portal1", text)
 
 
-@hook.command('portal1music')
-@hook.command
-def portalmusic(inp):
-    """portalmusic <title> - Look up Portal music.
-    Example: .portalmusic still alive"""
-    return get_sound_info("portal1music", inp)
+@hook.command(["portalmusic", "portal1music"])
+def portalmusic(text):
+    """portalmusic <title> - Look up Portal music. Example: .portalmusic still alive"""
+    return get_sound_info("portal1music", text)
 
 
-@hook.command('tf2sound')
-@hook.command
-def tf2(inp):
-    """tf2 [who - ]<quote> - Look up TF2 quote.
-    Example: .tf2 may i borrow your earpiece"""
-    return get_sound_info("tf2", inp)
+@hook.command(["tf2sound", "tf2"])
+def tf2(text):
+    """tf2 [who - ]<quote> - Look up TF2 quote. Example: .tf2 may i borrow your earpiece"""
+    return get_sound_info("tf2", text)
 
 
 @hook.command
-def tf2music(inp):
-    """tf2music title - Look up TF2 music lyrics.
-    Example: .tf2music rocket jump waltz"""
-    return get_sound_info("tf2music", inp)
+def tf2music(text):
+    """tf2music title - Look up TF2 music lyrics. Example: .tf2music rocket jump waltz"""
+    return get_sound_info("tf2music", text)
-- 
2.0.0

